Source/Session.swift

```diff
@@ -174,37 +178,41 @@ open class Session {
                        parameters: Parameters? = nil,
                        encoding: ParameterEncoding = URLEncoding.default,
                        headers: HTTPHeaders? = nil,
+                       interceptor: RequestInterceptor? = nil,
                        to destination: DownloadRequest.Destination? = nil) -> DownloadRequest {
         let convertible = RequestConvertible(url: convertible,
                                              method: method,
                                              parameters: parameters,
                                              encoding: encoding,
                                              headers: headers)
 
-        return download(convertible, to: destination)
+        return download(convertible, interceptor: interceptor, to: destination)
     }
 
     open func download<Parameters: Encodable>(_ convertible: URLConvertible,
                                               method: HTTPMethod = .get,
                                               parameters: Parameters? = nil,
                                               encoder: ParameterEncoder = JSONParameterEncoder.default,
                                               headers: HTTPHeaders? = nil,
+                                              interceptor: RequestInterceptor? = nil,
                                               to destination: DownloadRequest.Destination? = nil) -> DownloadRequest {
         let convertible = RequestEncodableConvertible(url: convertible,
                                                       method: method,
                                                       parameters: parameters,
                                                       encoder: encoder,
                                                       headers: headers)
 
-        return download(convertible, to: destination)
+        return download(convertible, interceptor: interceptor, to: destination)
     }
 
     open func download(_ convertible: URLRequestConvertible,
+                       interceptor: RequestInterceptor? = nil,
                        to destination: DownloadRequest.Destination? = nil) -> DownloadRequest {
```

Comment:
Same challenge here with the external API. I chose to put the `interceptor` in front of the `destination` to be consistent with the previous API. Again, very open to suggestions.